vectorizers/_vectorizers.py

```diff
@@ -899,6 +926,86 @@ def transform(self, X):
         return cooccurrences
 
 
+class LabeledTreeCooccurrenceVectorizer(BaseEstimator, TransformerMixin):
+    """
+    This takes a sequence of labeled directed graphs and produces a co-occurence count matrix
+    of their labels.
+
+    """
+
+    def __init__(
+        self,
+        token_dictionary=None,
+        min_occurrences=None,
+        max_occurrences=None,
+        min_frequency=None,
+        max_frequency=None,
+        ignored_tokens=None,
+        excluded_token_regex=None,
+        # window_function="fixed",
+        kernel_function="flat",
+        window_radius=5,
+        window_orientation="symmetric",
+        validate_data=True,
+    ):
+        self.token_dictionary = token_dictionary
+        self.min_occurrences = min_occurrences
+        self.min_frequency = min_frequency
+        self.max_occurrences = max_occurrences
+        self.max_frequency = max_frequency
+        self.ignored_tokens = ignored_tokens
+        self.excluded_token_regex = excluded_token_regex
+
+        # self.window_function = window_function
+        self.kernel_function = kernel_function
+        self.window_radius = window_radius
+
+        self.window_orientation = window_orientation
+        self.validate_data = validate_data
+
+    def fit(self, X, y=None, **fit_params):
+        """
+        Takes a sequence of labelled trees and learns the weighted co-occurence counts of the labels.
+        Parameters
+        ----------
+        X: sequence of (sparse.csr_matrix, label_to_index_dictionary) tuples
+            Here the sparse.csr_matrix should the adjacency matrix of one of your trees
+            The label_to_index_dictionary should be a dictionary from your vertex labels to row and column indices.
+
+        Returns
+        -------
+        self
+        """
+
+        # Pre-process trees.  This filters the tokens and learns a global token dictionary.
+        (
+            sequence_of_trees,
+            self.column_label_dictionary_,
+            self.column_index_dictionary_,
+            self._token_frequencies_,
+        ) = preprocess_tree_sequences(X)
```

Comment:
I take it this is in the "still to be written" category?

Reply:
Yep, still writing that.  I was mocking up the functions we would need and what they would take and return.  I thought it might be handy.